vaadin-components-gwt/src/main/java/com/vaadin/components/grid/GridComponent.java

```diff
@@ -439,7 +443,7 @@ public void sort(SortEvent<Object> event) {
             jsSort.push(sortOrder);
         }
         $(container).trigger("sort");
-        refresh();
+        refresh(false);
```

Comment:
Works for me (tried manually and programmatically) and all the tests seem to pass.